contrib/node/src/python/pants/contrib/node/subsystems/node_distribution.py

```diff
@@ -233,3 +271,38 @@ def yarnpkg_command(self, args, node_paths=None):
     """
     return self._command_gen(
       [self.install_node, self.install_yarnpkg], 'yarnpkg', args=args, node_paths=node_paths)
+
+  def _configure_eslinter(self, bootstrapped_support_path):
+    logger.debug('Copying {setupdir} to bootstrapped dir: {support_path}'
+                           .format(setupdir=self.eslint_setupdir, support_path=bootstrapped_support_path))
+    safe_rmtree(bootstrapped_support_path)
+    shutil.copytree(self.eslint_setupdir, bootstrapped_support_path)
+    return True
+
+  def fetch_eslint_supportdir(self):
+    """ Returns the path where the ESLint is bootstrapped.
+
+    :returns: The path where ESLint is bootstrapped and whether or not it is configured
+    :rtype: (string, bool)
+    """
+    bootstrapped_support_path = os.path.join(self.pants_workdir, 'lint', 'javascriptstyle')
+
+    # If the eslint_setupdir is not provided or missing required files, then
+    # clean up the directory so that Pants can install a pre-defined eslint version later on.
+    # Otherwise, if there is no configurations changes, rely on the cache.
+    # If there is a config change detected, use the new configuration.
+    configured = False
+    if self.eslint_setupdir:
+      configured = self._binary_util.is_bin_valid(self.eslint_setupdir,
+                                           [BinaryUtil.BinaryFileSpec('package.json'),
+                                            BinaryUtil.BinaryFileSpec('yarn.lock')])
+    if not configured:
+      safe_mkdir(bootstrapped_support_path, clean=True)
+    else:
+      binary_file_specs = [
+        BinaryUtil.BinaryFileSpec(f, hash_file(os.path.join(self.eslint_setupdir, f)))
+        for f in ['yarn.lock', 'package.json']]
+      installed = self._binary_util.is_bin_valid(bootstrapped_support_path, binary_file_specs)
```

Comment:
Because you're not persisting these fingerprints anywhere in association with the current options, they're always going to be re-calculated? And its not obvious where a good place to persist them would be.

I think that rather than bootstrapping these tools in a global location under pantsd, you should have the subsystem bootstrap them under the task's workdir... ie, give this method a signature:
`def eslint_supportdir(self, task_workdir)`. Because Tasks have fingerprints based on their options, you could avoid doing any of your own fingerprinting, and instead just check for existence of the files.

When the Task fingerprint changes, it gets a brand new directory, and would re-bootstrap as you would expect.

Reply:
@stuhood: I am not sure I fully understand the concept of fingerprinting. Can you clarify what you mean by:
- "Because Tasks have fingerprints based on their options, you could avoid doing any of your own fingerprinting, and instead just check for existence of the files."

If the configuration changes, wouldn't the task options remain the same?